core/src/main/scala/tech/beshu/ror/accesscontrol/factory/decoders/definitions/LdapServicesDecoder.scala

```diff
@@ -76,8 +76,8 @@ object LdapServicesDecoder {
   private def decodeLdapService(cursor: HCursor)
                                (implicit ldapConnectionPoolProvider: UnboundidLdapConnectionPoolProvider): Task[Either[DecodingFailure, LdapUserService]] = {
     for {
-      authenticationService <- (authenticationServiceDecoder: AsyncDecoder[LdapAuthenticationService])(cursor)
-      authortizationService <- (authorizationServiceDecoder: AsyncDecoder[LdapAuthorizationService])(cursor)
+      authenticationService <- (circuitBreakerAuthenticationServiceDecoder: AsyncDecoder[LdapAuthenticationService])(cursor)
+      authortizationService <- (circuitBreakerAuthorizationServiceDecoder: AsyncDecoder[LdapAuthorizationService])(cursor)
     } yield (authenticationService, authortizationService) match {
       case (Right(authn), Right(authz)) => Right {
         new ComposedLdapAuthService(authn.id, authn, authz)
```

Comment:
I'm not sure this is a right level to apply the CB decorator. Please take a look at Caching decorator. 

In `decodeLdapService` we create an LDAP service (which can be authn, authz or authn&z/composed). Now, in case of authn&z service, we have two CBs applied and then composition. But IMO we should first compose authn and authz services into authn&z service and then apply (only one) CB at the top of the composition.